mdtk/degradations.py

```diff
@@ -91,9 +99,11 @@ def pre_process(df, sort=False):
     df : pd.DataFrame
         The postprocessed dataframe.
     """
+    df = df.loc[:, NOTE_DF_SORT_ORDER]
```

Comment:
(RE: #37)

There is an argument for removing this line:
- Nothing would err with too many columns. Nor would any existing columns be removed. Rather, added notes would just have NaNs in those columns, which is fine.
- This would cause the resulting dfs to have floats instead of ints if a new note is added (since NaN can't be an int).
- Some degradations (remove_note, shift_XXX, join_notes) would still work fine with too many columns.
- This would cause dfs with too few columns to work for remove_note (still fail on the others).

Reply:
Why did we put it there in the first place? Is there somewhere that insists on a column order (presume so!).

Reply:
This line also removes additional columns, which would break some degradations. The reordering is just a happy accident.

Reply:
Ah yes, I see. I think in another module I wanted the dataframes to have a specific set of columns and in a specific order when they were first created, pre-degradation. I don't see a reason for this restriction in your pre_process function for degradations here. I think you're good to remove and see if it breaks any tests!

Reply:
It will break this test I added: https://github.com/JamesOwers/midi_degradation_toolkit/pull/86#discussion_r341009252

But no others. Although that's only because we never test with incorrect columns.

The point with having this line is for if people make their own dfs, but I think it's fine to assume they have a reasonable set, and weird things will happen if not (eg. Nan's and floats appear)

Reply:
I can't change this easily now from Switzerland